pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -631,168 +631,146 @@ protected void handleProducerRemoved(Producer producer) {
                                                  Map<String, String> metadata, boolean readCompacted,
                                                  InitialPosition initialPosition,
                                                  long startMessageRollbackDurationSec,
-                                                 boolean replicatedSubscriptionState,
+                                                 boolean replicatedSubscriptionStateArg,
                                                  KeySharedMeta keySharedMeta) {
+        return brokerService.checkTopicNsOwnership(getName()).thenCompose(__ -> {
+            if (readCompacted && !(subType == SubType.Failover || subType == SubType.Exclusive)) {
```

Comment:
we could move this validation outside of the `checkTopicNsOwnership` block.
not a big deal, it is something that does not happen in the happy path

Reply:
Done